### PR TITLE
Ref [Task #49906] Search Film

### PR DESCRIPTION
**Pull request by haihn-3019**
https://edu-redmine.sun-asterisk.vn/issues/49906
image

### CODE REVIEW

**Comment by haihn-3019** [comment]
ready

File: src/store/film/film.action.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/* eslint-disable import/prefer-default-export */
+export function handleChangeDate(state, { payload: dateSelected }) {
+  const dateTimeSelected = state.dateTimes.find((dateTime) => dateTime.date === dateSelected);
+
+  const { times } = dateTimeSelected;
```
Comment (vietnq-1097) [inline src/store/film/film.action.js]:
```suggestion
  const { times } = state.dateTimes.find((dateTime) => dateTime.date === dateSelected);


```

File: src/store/film/film.slice.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  },
+  reducers: {
+    changeDate: handleChangeDate
+  },
+  extraReducers: {
+    [getFilmByID.fulfilled]: (state, { payload }) => {
+      state.isLoading = false;
+      state.dateSelected = '';
+      const { id, dateTimes, info } = payload;
+      state.id = id;
```
Comment (vietnq-1097) [inline src/store/film/film.slice.js]:
```suggestion
      const { id, dateTimes, info } = payload;

      state.isLoading = false;
      state.dateSelected = '';
      state.id = id;

```

File: src/store/film/film.thunk.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+// eslint-disable-next-line import/prefer-default-export
+export const getFilmByID = createAsyncThunk('get/getFilmByID', async (id, { rejectWithValue }) => {
+  try {
+    const res = await axios.get(`${process.env.BASE_URL}/films-showing/${id}`);
+    console.log(res, 'res filmByID');
+    return res.data;
```
Comment (vietnq-1097) [inline src/store/film/film.thunk.js]:
```suggestion
    const { data } = await axios.get(`${process.env.BASE_URL}/films-showing/${id}`);

    return data;

```

**Pull request merged**